code/modules/food&drinks/drinks/drinks.dm

```diff
@@ -24,7 +24,7 @@
 	return
 
 /obj/item/weapon/reagent_containers/food/drinks/attack(mob/M as mob, mob/user as mob, def_zone)
-
+	..()
```

Comment:
It calls the parent's attack proc, which contains the code to deal with heated items.